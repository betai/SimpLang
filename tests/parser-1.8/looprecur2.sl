    loop a = 100 and
		 b = 0
	in
	    if (a == 0) then
	        b
		else
			recur ((a+-1)) ((b+1))
		end
	end
