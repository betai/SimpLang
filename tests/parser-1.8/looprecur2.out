loop
    a
      100
    b
      0
  if
    ==
      a
      0
    b
    recur
      +
        a
        -
          1
      +
        b
        1
